File: leetsync/2766-find-the-prefix-common-array-of-two-arrays/find-the-prefix-common-array-of-two-arrays.java
class Solution {
    public int[] findThePrefixCommonArray(int[] a, int[] b) {
        int size = a.length;
        int[] result = new int[size];
        int[] visited = new int[size + 1];
        int currentCount = 0;
        for (int i = 0; i < size; i++) {
            if (++visited[a[i]] == 2) {
                currentCount++;
            }
            if (++visited[b[i]] == 2) {
                currentCount++;
            }
            result[i] = currentCount;
        }
        return result;
    }
}

File: leetsync/2766-find-the-prefix-common-array-of-two-arrays/find-the-prefix-common-array-of-two-arrays.go
func findThePrefixCommonArray(a []int, b []int) []int {
	n := len(a)
	result := make([]int, n)
	visited := make([]int, n+1)
	currentCount := 0
	for i := 0; i < n; i++ {
		visited[a[i]]++
		if visited[a[i]] == 2 {
			currentCount++
		}
		visited[b[i]]++
		if visited[b[i]] == 2 {
			currentCount++
		}
		result[i] = currentCount
	}
	return result
}